### PR TITLE
Guarin lig 3054 add byol imagenet resnet50 benchmark

### LINKED ISSUE #1287: BYOL

### PR DESCRIPTION
**Pull request by guarin**
## Changes

* Add BYOL ImageNet ResNet50 benchmark

Closes #1287 

## Results

First run gave a bit lower results than expected:
```
max val_online_cls_top1: 0.5502600073814392
max val_online_cls_top5: 0.7950400114059448
knn val_top1: 0.3464199900627136
knn val_top5: 0.6211400032043457
max linear val_top1: 0.5135800242424011
max linear val_top5: 0.761900007724762
max finetune val_top1: 0.731939971446991
max finetune val_top5: 0.9126600027084351
```
Figured out that the BYOL reference implementation uses a symmetric loss:
```
loss = loss_fn(teacher_view_0, student_view_1) + loss_fn(teacher_view_1, student_view_2)
```
and just sums the two terms instead of averaging. So loss and gradients were off by factor of 2 because I took the average of the two losses.

## New Results

Looks much better now :)

```
max val_online_cls_top1: 0.6177999973297119
max val_online_cls_top5: 0.8418599963188171
knn val_top1: 0.45649999380111694
knn val_top5: 0.7477800250053406
max linear val_top1: 0.6240599751472473
max linear val_top5: 0.8471800088882446
max finetune val_top1: 0.7399399876594543
max finetune val_top5: 0.9189199805259705
```

The paper sadly doesn't report results for 100ep training but we are in the range of our SimCLR with sqrt learning reate results. Our byol linear eval results are a bit lower than those reported in the solo-learn package but they use different evaluation settings, so I would assume the biggest difference comes from there.

### CODE REVIEW

**Comment by IgorSusmelj** [comment]
max finetune val_top1: 0.731939971446991                                  
max finetune val_top5: 0.9126600027084351
max finetune val_top1: 0.731939971446991
max finetune val_top5: 0.9126600027084351

File: README.md [hunk 1]
```diff
@@ -287,15 +287,17 @@ tuned for maximum accuracy. For detailed results and more info about the benchma
 > 
 > See the [benchmarking scripts](./benchmarks/imagenet/resnet50/) for details.
 
-| Model       | Backbone | Batch Size | Epochs | Linear Top1 | Finetune Top1 | KNN Top1 | Tensorboard | Checkpoint |
-|-------------|----------|------------|--------|-------------|---------------|----------|-------------|------------|
-| DINO        | Res50    |        128 |    100 |        68.2 |          72.5 |     49.9 |      [link](https://tensorboard.dev/experiment/DvKHX9sNSWWqDrRksllPLA) |       [link](https://lightly-ssl-checkpoints.s3.amazonaws.com/imagenet_resnet50_dino_2023-06-06_13-59-48/pretrain/version_0/checkpoints/epoch%3D99-step%3D1000900.ckpt) |
```
Comment (guarin) [inline README.md:292]: Sorted rows alphabetically

**Review by IgorSusmelj — APPROVED** [review]
